velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include "velox/functions/Udf.h"
+
+namespace facebook::velox::functions {
+
+template <bool isMax, typename VeloxType>
+VELOX_UDF_BEGIN(ArrayMinMaxFunction)
+
+template <typename T>
+void update(T& currentValue, const T& candidateValue) {
+  if constexpr (isMax) {
+    if (candidateValue > currentValue) {
+      currentValue = candidateValue;
+    }
+  } else {
+    if (candidateValue < currentValue) {
+      currentValue = candidateValue;
+    }
+  }
+}
+
+void assign(out_type<VeloxType>& out, const arg_type<VeloxType>& value) {
+  if constexpr (std::is_same<Varchar, VeloxType>::value) {
+    // TODO: reuse strings once support landed.
+    out.resize(value.size());
+    std::memcpy(out.data(), value.data(), value.size());
+  } else {
+    out = value;
+  }
+}
+
+FOLLY_ALWAYS_INLINE bool call(
+    out_type<VeloxType>& out,
+    const arg_type<Array<VeloxType>>& array) {
+  // Result is null if array is empty.
+  if (array.size() == 0) {
+    return false;
+  }
+
+  if (!array.mayHaveNulls()) {
+    // Input array does not have nulls.
+    auto currentValue = *array[0];
```

Comment:
curious why do you have to take * here, doesnt array[] store the types by value ?